### PR TITLE
Handling lack of ffmpeg or imagemagick.

### PR DESCRIPTION
**Pull request by witekha3**

### CODE REVIEW

File: src/third_party/ffmpeg.py
Comment (nekoffski) [inline src/third_party/ffmpeg.py:1]: It's not a 3rdparty code, it checks for 3rdparty. So, I'd move it to the source directory.

File: src/third_party/ffmpeg.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import subprocess
+import logging
+
+
+def check_ffmpeg_installed():
```
Comment (nekoffski) [inline src/third_party/ffmpeg.py]: is_ffmpeg_installed

File: src/third_party/imagemagick.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import subprocess
+import logging
+
+
+def check_imagemagick_installed():
```
Comment (nekoffski) [inline src/third_party/imagemagick.py]: is_imagemagick_installed

File: src/third_party/ffmpeg.py (hunk 1 above)
Comment (nekoffski) [reply src/third_party/ffmpeg.py]: Also, it seems like twose functions could be merged into some generic code with parameters.

File: src/sub_movie_generator.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+from src.third_party.ffmpeg import check_ffmpeg_installed
+from src.third_party.imagemagick import check_imagemagick_installed
+
+
+class SubMovieGenerator:
+
+    def __init__(self, movie_path: Path, model_size: str, lang: Optional[str] = None):
+        if not check_ffmpeg_installed() or not check_imagemagick_installed():
+            exit()
```
Comment (nekoffski) [inline src/sub_movie_generator.py]: Having multiple exit points is generally not a great idea, I'd throw an exception here and catch it in the main function.

Comment (nekoffski) [reply src/sub_movie_generator.py]: Especially in constructor.

File: src/sub_movie_generator.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        self.video = Video(movie_path)
```
Comment (nekoffski) [inline src/sub_movie_generator.py:12]: Some variables are underscored whether some aren't. What is the convention?

**Review by nekoffski — CHANGES_REQUESTED** [review]

File: src/sub_movie_generator.py (hunk 4 above)
Comment (witekha3) [reply src/sub_movie_generator.py:12]: https://stackoverflow.com/a/1301409

File: src/third_party/ffmpeg.py
Comment (witekha3) [reply src/third_party/ffmpeg.py:1]: You're right

File: src/sub_movie_generator.py (hunk 4 above)
Comment (nekoffski) [reply src/sub_movie_generator.py:12]: Yup, I get this but what's with the not underlined variables? Is video accessed outside of the class?

Comment (witekha3) [reply src/sub_movie_generator.py:12]: I thought that there might be a situation that these values will be used someday. I can set all of them to private if you would like to

Comment (nekoffski) [reply src/sub_movie_generator.py:12]: If you think so let's let them as it is.

**Review by nekoffski — APPROVED** [review]

**Pull request merged**